### PR TITLE
CPLAT-8341: Post-Dart2 cleanup

### PR DESCRIPTION
**Pull request by olesiathoms-wk**
## Problem/Solution
We can remove dart 1
### Changes
- 'dartfmt` fixes
- remove `dart2_constant`
- update dependencies
- cleanup `.gitignore`
- use base image
## Testing instructions
- [ ] CI should be sufficient

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
The items listed below may not capture all security relevant changes. Before providing a security review, be sure to review the entire PR for security impact.
<details>
<summary>(1) Security relevant changes were detected</summary>
<li>Watched file <code>lib/src/transformed_source_file.dart</code> modified</li>
</details>

## Action Items
* Obtain a [security review](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); reviewer should pay special attention to insights listed above
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
+addons:
+  chrome: stable
+
+jobs:
+  include:
+    - dart: 2.4.0
+      name: "SDK: 2.4.0"
+      script:
+        - pub run dart_dev analyze
+        - pub run dart_dev test --release
```
Comment (todbachman-wf) [inline .travis.yml]: Since we only run the unit tests on `vm`, we probably don't need to run them with the `--release` flag.

File: pubspec.yaml [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
-    dart_style: ^1.0.9
-    dependency_validator: ^1.1.0
-    test: ^1.5.3
-
-environment:
-    sdk: ">=2.0.0 <3.0.0"
+  build_runner: ^1.7.1
+  build_test: ^0.10.9
+  build_vm_compilers: ^1.0.3
+  build_web_compilers: ^2.5.1
```
Comment (todbachman-wf) [inline pubspec.yaml]: Do we need all of these `build_*` deps? It looks like we didn't have them before.

File: tool/dart_dev/config.dart [hunk 3]
```diff
@@ -12,19 +12,6 @@
...
-  List<String> directories = ['lib/', 'test/', 'tool/'];
-
-  config.analyze.entryPoints = ['lib/', 'test/', 'test/unit/', 'tool/'];
-  config.copyLicense.directories = directories;
-  config.format.paths = directories;
-  config.test.unitTests = ['test/unit/'];
-
-  await dev(args);
-}
+final Map<String, DevTool> config = coreConfig;
```
Comment (todbachman-wf) [inline tool/dart_dev/config.dart]: Since we just use the `coreConfig` unchanged, we could just delete this file entirely.

File: .travis.yml (hunk 1 above)
Comment (olesiathoms-wk) [reply .travis.yml]: why not? is there no difference between them for `vm`?

File: pubspec.yaml (hunk 2 above)
Comment (olesiathoms-wk) [reply pubspec.yaml]: I believe it's being added automatically with `d1down_update_pubspec`, but you right we can remove them

File: .travis.yml (hunk 1 above)
Comment (todbachman-wf) [reply .travis.yml]: Yeah, it's meaningless in the context of the vm.

**Comment by evanweible-wf** [comment]
QA +1
- [x] CI passes

@Workiva/release-management-p

**Comment by evanweible-wf** [comment]
+1 security
- Changes to flagged file are non-functional

**Pull request merged**